File: 2025-11-30.js
function detectAI(text) {
    let dashCount = countDashes(text)
    let parethesesCount = countParentheses(text)
    let bigWordsCount = countBigWords(text)

    if (dashCount >= 2) {
        return "AI"
    }
    if (parethesesCount >= 2) {
        return "AI"
    }
    if (bigWordsCount >= 3) {
        return "AI"
    }

    return "Human";
}

function countDashes(text) {
    let count = 0

    for (let char of text) {
        if (char === '-') {
            count++
        }
    }

    return count;
}

function countParentheses(text) {
    let count = 0
    let open_count = 0

    for (let char of text) {
        if (char === '(') {
            open_count++
        }
        else if (char === ')') {
            if (open_count > 0) {
                count++
                open_count--
            }
        }
    }

    return count;
}

function countBigWords(text) {
    let count = 0
    let word = []

    for (let char of text) {
        if (char !== ' ') {
            word.push(char);
        }
        else {
            let length = word.length;
            if (length >= 7) {
                count++;
            }

            word = [];
        }
    }

    return count;
}




// TEST 1 - return "Human"
let result1 = detectAI("The quick brown fox jumped over the lazy dog.")
console.log(result1);

// TEST 2 - return "Human"
let result2 = detectAI("The hypersonic brown fox - jumped (over) the lazy dog.")
console.log(result2);

// TEST 3 - return "AI"
let result3 = detectAI("Yes - you're right! I made a mistake there - let me try again.")
console.log(result3);

// TEST 4 - return "AI"
let result4 = detectAI("The extraordinary students were studying vivaciously.")
console.log(result4);

// TEST 5 - return "AI"
let result5 = detectAI("The (excited) student was (coding) in the library.")
console.log(result5);

File: Python/2025_11_30.py
def detect_ai(text):
    dash_count = count_dashes(text)
    parenthesis_count = count_parentheses(text)
    big_words_count = count_big_words(text)

    if dash_count >= 2:
        return "AI"

    if parenthesis_count >= 2:
        return "AI"

    if big_words_count >= 3:
        return "AI"

    return "Human"

def count_dashes(s):
    count = 0

    for char in s:
        if char == '-':
            count += 1

    return count

def count_parentheses(s):
    count = 0
    open_count = 0

    for char in s:

        if char == '(':
            open_count += 1

        elif char == ')':
            if open_count > 0:
                count += 1
                open_count -= 1

    return count

def count_big_words(s):
    count = 0
    word = []

    for char in s:
        if char != ' ':
            word.append(char)
        else:
            length = len(word)
            if length >= 7:
                count += 1

            word = []

    return count





# TEST 1 - return "Human"
print(detect_ai("The quick brown fox jumped over the lazy dog."))

# TEST 2 - return "Human"
print(detect_ai("The hypersonic brown fox - jumped (over) the lazy dog."))

# TEST 3 - return "AI"
print(detect_ai("Yes - you're right! I made a mistake there - let me try again."))

# TEST 4 - return "AI"
print(detect_ai("The extraordinary students were studying vivaciously."))

# TEST 5 - return "AI"
print(detect_ai("The (excited) student was (coding) in the library."))